lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -280,13 +301,158 @@ impl Client {
+                                    }
+                                } else if rdp_req
+                                    .create_options
+                                    .contains(flags::CreateOptions::FILE_DIRECTORY_FILE)
+                                {
+                                    // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L237
+                                    // ERROR_DIRECTORY --> STATUS_NOT_A_DIRECTORY: https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_main.c#L118
+                                    return cli.prep_device_create_response(
+                                        &rdp_req,
+                                        NTSTATUS::STATUS_NOT_A_DIRECTORY,
+                                    );
+                                }
+                            } else if res.err_code == 2 {
+                                // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L242
+                                if rdp_req
+                                    .create_options
+                                    .contains(flags::CreateOptions::FILE_DIRECTORY_FILE)
+                                {
+                                    if rdp_req.create_disposition.intersects(
+                                        flags::CreateDisposition::FILE_OPEN_IF
+                                            | flags::CreateDisposition::FILE_CREATE,
+                                    ) {
+                                        // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L252
+                                        return cli.tdp_sd_create(rdp_req, 1);
+                                    } else {
+                                        // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L258
+                                        // ERROR_FILE_NOT_FOUND --> STATUS_NO_SUCH_FILE: https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_main.c#L85
+                                        return cli.prep_device_create_response(
+                                            &rdp_req,
+                                            NTSTATUS::STATUS_NO_SUCH_FILE,
+                                        );
+                                    }
+                                }
+                            } else {
+                                // generic RDP error
+                                return cli.prep_device_create_response(
+                                    &rdp_req,
+                                    NTSTATUS::STATUS_UNSUCCESSFUL,
+                                );
+                            }
```

Comment:
This has been refactored into a `match` statement on a `TdpErrCode` enum in https://github.com/gravitational/teleport/pull/12830

Reply:
Ok, will take a closer look in the next PR then as this is difficult to follow